source/security.txt

```diff
@@ -0,0 +1,152 @@
+
+.. tip::
+
+   To learn more about any of the mechanisms shown on this page, see the link
+   provided in each section.
+
+To use an authentication example from this page, copy the code example into the
+:ref:`sample application <c-auth-sample>` or your own application.
+Make sure to replace all placeholders in the code examples, such as ``<hostname>``, with
+the relevant values for your MongoDB deployment.
+
+.. _c-auth-sample:
+
+.. include:: /includes/usage-examples/sample-app-intro.rst
+
+.. literalinclude:: /includes/usage-examples/connect-sample-app.c
+   :language: c
+   :copyable: true
+   :linenos:
+   :emphasize-lines: 14-16
+
+SCRAM-SHA-256
+-------------
+
+The following code shows how to authenticate by using the ``SCRAM-SHA-256``
+authentication mechanism:
+
+.. literalinclude:: /includes/security/authentication.c
+   :language: c
+   :copyable: true
+   :start-after: // start-scram-sha-256
+   :end-before: // end-scram-sha-256
+
+To learn more about SCRAM-SHA-256 authentication, see :ref:`c-scram-sha-256` in
+the Authentication guide.
+
+SCRAM-SHA-1
+-----------
+
+The following code shows how to authenticate by using the ``SCRAM-SHA-1``
+authentication mechanism:
+
+.. literalinclude:: /includes/security/authentication.c
+   :language: c
+   :copyable: true
+   :start-after: // start-scram-sha-1
+   :end-before: // end-scram-sha-1
+
+To learn more about SCRAM-SHA-1 authentication, see :ref:`c-scram-sha-1` in
+the Authentication guide.
+
+MONGODB X.509
+-------------
+
+The following code shows how to create a connection URI to authenticate by using
+the ``X.509`` authentication mechanism:
+
+.. literalinclude:: /includes/security/authentication.c
+   :language: c
+   :copyable: true
+   :start-after: // start-x509
+   :end-before: // end-x509
+
+To learn more about X.509 authentication, see :ref:`c-mongodb-x509` in
+the Authentication guide.
+
+MONGODB-AWS
+-----------
+
+The following sections show how to connect to MongoDB by using the ``MONGODB-AWS``
+authentication mechanism. When you use the ``MONGODB-AWS`` mechanism, the {+driver-short+}
+attempts to retrieve your AWS credentials from the following sources, in the order listed:
+
+1. Named parameters passed to the connection URI
+#. Environment variables
+#. AWS EKS AssumeRoleWithWebIdentity request
+#. ECS container metadata
+#. EC2 instance metadata
+ 
+Each section shows how to authenticate with ``MONGODB-AWS`` when retrieving your
+AWS credentials from a connection URI or the alternative external sources.
+
+To learn more about authenticating with AWS, see :ref:`c-mongodb-aws` in the
+Authentication guide.
+
+Connection URI
+~~~~~~~~~~~~~~
+
+The following code shows how to pass AWS credentials in a connection URI
+to authenticate with ``MONGODB-AWS``:
+
+.. literalinclude:: /includes/security/authentication.c
+   :language: c
+   :copyable: true
+   :start-after: // start-aws-connection-uri
+   :end-before: // end-aws-connection-uri
```

Comment:
Question: Are both code samples in the `MONGODB-AWS` section meant to be identical? Wondering if something maybe got overlooked in the earlier PR that contained the code that you're pulling from for this PR.

Reply:
Oh good catch, I don't think the code examples (other than the connection URI credentials one) should have the `<AWS IAM access key ID>:<AWS IAM secret access key>` placeholders. I'll remove them here and on the Authentication page